src/network/ClientConnection.ts

```diff
@@ -244,19 +244,15 @@ export class FragmentedClientMessageHandler {
         clientMessage.dropFragmentationFrame();
         if (fragmentationFrame.hasBeginFragmentFlag()) {
             this.fragmentedMessages.set(fragmentationId, clientMessage);
-        } else if (fragmentationFrame.hasEndFragmentFlag()) {
-            const mergedMessage = this.mergeIntoExistingClientMessage(fragmentationId, clientMessage);
-            callback(mergedMessage);
         } else {
-            this.mergeIntoExistingClientMessage(fragmentationId, clientMessage);
+            const existingMessage = this.fragmentedMessages.get(fragmentationId);
```

Comment:
Is it theoretically possible that a client didn't receive previous fragment and thus `existingMessage` is `undefined`?

Reply:
Yes, in fact, this is possible. I added a check before. If it has no begin part, It logs this with the fragmentation id (this is the only thing we could extract from middle or last fragments). Also, enhanced the test case with this scenario

Reply:
Thanks for the update. The change looks good to me.